tests/test_director.py

```diff
@@ -694,6 +747,35 @@ def test_40_add_target_for_ecu(self):
 
 
 
+  def test_60_register_vehicle(self):
```

Comment:
Consider testing for invalid vin arguments.  I noticed there is a `_check_registration_is_sane()` function, so you might as well test it.  
Also, why do you use an assert in `_check_registration_is_sane` rather than a specific exception?

Reply:
A FormatError is raised if the given VIN is not a the correct format. If, however, the invariants are broken on the data structure, that is a result of an error in the code (or, I suppose, someone tinkering directly with the contents of inventorydb), not an error related to any input or use of the class. That struck me as a candidate for use of an assertion rather than an exception. I'm not confident that that's the right policy, though. What do you think I should raise when the code is observed to be broken?

Reply:
I thought `_check_registration_is_sane()` was validating the arguments.  I guess having a test for invalid arguments and checking against FormatError is fine.

Reply:
It does, but it also checks to see if situations that should be impossible have arisen to detect coding errors (e.g. an ostensibly registered VIN appears in only one of the X>1 places you'd expect a registered VIN to appear)

Reply:
But anyway, I'm still not sure asserts are the right way to do that. What do you think?

Reply:
I recommend keeping the assert if it's truly an impossible to reach internal error, and separating the other validation checks (so that we can test for them via exceptions).  We should also limit the number of asserts used throughout the code, if possible.

Reply:
Thanks. I'll add a test for FormatError.